plugin-server/src/cdp/hog-transformations/hog-transformer.service.ts

```diff
@@ -132,6 +133,40 @@ export class HogTransformerService {
                 // For now, execute each transformation function in sequence
                 for (const hogFunction of teamHogFunctions) {
                     const transformationIdentifier = `${hogFunction.name} (${hogFunction.id})`
+
+                    // Check if we should apply this transformation based on its filters
+                    if (this.hub.FILTER_TRANSFORMATIONS_ENABLED) {
+                        const globals = this.createInvocationGlobals(event)
+                        const filterGlobals = convertToHogFunctionFilterGlobal(globals)
+                        const shouldApplyTransformation = this.checkEventAgainstFilters(hogFunction, filterGlobals)
+
+                        if (!shouldApplyTransformation) {
+                            logger.info('🔍', 'Skipping transformation due to filter mismatch', {
+                                event_id: event.uuid,
+                                transformation_id: hogFunction.id,
+                                transformation_name: hogFunction.name,
+                            })
+                            // Add a log entry to the results when filtered out
+                            const globalsWithInputs = buildGlobalsWithInputs(globals, {
+                                ...(hogFunction.inputs ?? {}),
+                                ...(hogFunction.encrypted_inputs ?? {}),
+                            })
```

Comment:
This is a lot of cpu cycles for something that won't be used. Given that we only need this for pushing a result for logging wouldn't it be eaiser to just create a dummy object?